Core/Clauses/QuerySourceMapping.cs

```diff
@@ -76,5 +76,18 @@ public Expression GetExpression (IQuerySource querySource)
 
       return expression;
     }
+
+    public void RemoveMapping (IQuerySource querySource)
```

Comment:
Please move it up to ReplaceMapping so we keep the modification-methods grouped together.